gemma.h

```diff
@@ -75,9 +72,8 @@ struct GemmaInterface;
 
 struct Gemma {
   Gemma(const Path& tokenizer_path, const Path& compressed_weights_path,
-        const Path& weights_path, Model model_type, hwy::ThreadPool& pool);
-  Gemma(const Path& tokenizer_path, const Path& compressed_weights_path,
-        Model model_type, hwy::ThreadPool& pool);
+        const Path& weights_path, Model model_type, ModelTraining training,
```

Comment:
Don't have to tackle it in this PR, but we should consider consolidating Model and ModelTraining.

The issue is that as we add more variations, it doesn't always make sense to assume there's the cartesian product of all combinations, in which case having the ModelTraining metadata as part of the Model type might make more sense.

Reply:
Now that the `experimental` changes were merged into `dev` (and `main`), you can go back to targeting PRs directly to `dev` (which will jan's commits).

Reply:
OK, I'll do this after moving the conditional branch out of the pos2 loop later.